packages/events/src/EventBoundary.ts

```diff
@@ -1382,6 +1386,7 @@ export class EventBoundary
                 i < j && !e.propagationImmediatelyStopped;
                 i++)
             {
+                if (listeners[i].once) e.currentTarget.removeListener(type, listeners[i].fn, undefined, true);
```

Comment:
Does `removeListener` modify the listeners length? If so, this could potentially skip over listeners:
```js
target.on('click', () => {})
target.once('click', () => {})
target.on('click', () => {}) // <-- skipped?
```

Reply:
No, `removeListener` create a new array instead of modifying the original one (see the original code <https://github.com/primus/eventemitter3/blob/b99e3efb9d62392c7c7275d2f98d3d9dbc9b368f/index.js#L244-L292>).

Reply:
The original [`EventEmitter#emit`](https://github.com/primus/eventemitter3/blob/b99e3efb9d62392c7c7275d2f98d3d9dbc9b368f/index.js#L159-L216) do this the same way:

```js
    if (listeners.once) this.removeListener(event, listeners.fn, undefined, true);

    switch (len) {
      case 1: return listeners.fn.call(listeners.context), true;
      case 2: return listeners.fn.call(listeners.context, a1), true;
      case 3: return listeners.fn.call(listeners.context, a1, a2), true;
      case 4: return listeners.fn.call(listeners.context, a1, a2, a3), true;
      case 5: return listeners.fn.call(listeners.context, a1, a2, a3, a4), true;
      case 6: return listeners.fn.call(listeners.context, a1, a2, a3, a4, a5), true;
    }

    for (i = 1, args = new Array(len -1); i < len; i++) {
      args[i - 1] = arguments[i];
    }

    listeners.fn.apply(listeners.context, args);
```